sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -78,7 +80,8 @@ case class CreateTableAsSelectExec(
     }
 
     Utils.tryWithSafeFinallyAndFailureCallbacks({
-      catalog.createTable(ident, query.schema, partitioning.toArray, properties.asJava) match {
+      catalog.createTable(
+        ident, query.schema, partitioning.toArray, properties.asJava) match {
```

Comment:
should the schema be called with `asNullable`? I feel the nullability of the input data cannot be trusted (or just because a column is not-nullable for a query doesn't mean it can't be null in the future)

Reply:
I'm not quite willing to adjust any assumptions from the previous implementation in this PR - that can be a follow-up question perhaps for the dev list or in another JIRA.

Reply:
I agree. The scope of this suggestion is larger than this PR.